test/unit/modules/cache.js

```diff
@@ -0,0 +1,393 @@
+					expect(res).to.eql(value);
+					done(err, value);
+				});
+			});
+		});
+	});
+
+	describe('flushDb', function () {
+
+		it('should remove all keys from cache', function (done) {
+			var key1 = 'test_key1';
+			var key2 = 'test_key2';
+			var dummyValue = { a: 'dummyValue' };
+			async.series([
+				// save new entries in cache
+				function (callback) {
+					async.map([key1, key2], function (key, cb) {
+						cache.setJsonForKey(key, dummyValue, cb);
+					}, function (err, result) {
+						expect(err).to.not.exist;
+						expect(result).to.be.an('array');
+						callback(err, result);
+					});
+				},
+				// flush cache database
+				function (callback) {
+					cache.flushDb(function (err, status) {
+						expect(err).to.not.exist;
+						expect(status).to.equal('OK');
+						callback(err, status);
+					});
+				},
+				// check if entries exist
+				function (callback) {
+					async.map([key1, key2], function (key, cb) {
+						cache.getJsonForKey(key, cb);
+					}, function (err, result) {
+						expect(err).to.not.exist;
+						expect(result).to.be.an('array');
+						expect(result).to.have.length(2);
+						result.forEach(function (value) {
+							expect(value).to.eql(null);
+						});
+						callback(err, result);
+						done(err, result);
+					});
+				}]
+			);
+		});
+	});
+
+	describe('removeByPattern', function () {
+
+		it('should remove keys matching the pattern', function (done) {
+			var key = '/api/transactions?123';
+			var value = {testObject: 'testValue'};
+			var pattern = '/api/transactions*';
+
+			cache.setJsonForKey(key, value, function (err, status) {
+				expect(err).to.not.exist;
+				expect(status).to.equal('OK');
+				cache.removeByPattern(pattern, function (err) {
+					expect(err).to.not.exist;
+					cache.getJsonForKey(key, function (err, res) {
+						expect(err).to.not.exist;
+						expect(res).to.equal(null);
+						done();
+					});
+				});
+			});
+		});
+
+		it('should not remove keys that dont match pattern', function (done) {
+			var key = '/api/transactions?123';
+			var value = {testObject: 'testValue'};
+			var pattern = '/api/delegate*';
+
+			cache.setJsonForKey(key, value, function (err, status) {
+				expect(err).to.not.exist;
+				expect(status).to.equal('OK');
+				cache.removeByPattern(pattern, function (err) {
+					expect(err).to.not.exist;
+					cache.getJsonForKey(key, function (err, res) {
+						expect(err).to.not.exist;
+						expect(res).to.eql(value);
+						done();
+					});
+				});
+			});
+		});
+
+	});
+	
+	describe('onNewBlock', function () {
+
+		it('should remove all keys matching pattern /api/transactions', function (done) {
+			var key = '/api/transactions?123';
+			var value = {testObject: 'testValue'};
+
+			cache.setJsonForKey(key, value, function (err, status) {
+				expect(err).to.not.exist;
+				expect(status).to.equal('OK');
+				//this function doesn't accept a callback, so waiting 2 seconds before checking if the actions were completed
```

Comment:
These functions are events listeners which are called with multiple arguments. Adding a callback would require accepting a few other parameters which I didn't need. So, I thought I'd just wait a while before running this test (since I don't need a callback in the rest of the app). I'll add the optional callback if you think that'd be a better approach.